test/e2e/cluster.go

```diff
@@ -161,7 +170,7 @@ func createStatefulSet(ctx context.Context, cli kubernetes.Interface) error {
 						AccessModes: []corev1.PersistentVolumeAccessMode{
 							corev1.ReadWriteOnce,
 						},
-						StorageClassName: to.StringPtr("managed-premium"),
+						StorageClassName: to.StringPtr(defaultStorageClass),
```

Comment:
how about we just remove this entirely. Is it critically important that we hard-code the default Storage Class name, or can we just let k8s to k8s things and use the default SC if none is specified? I feel like this E2E is more a test of "can the cluster create default storage?" vs "can the cluster create storage of a particular named type?" - I feel like in a multi-version world, saying "use the default" is the easiest path that tests what we are actually after.

Reply:
I was meaning to test this, thanks for the reminder.